File: 백준/Silver/2847. 게임을 만든 동준이/게임을 만든 동준이.py
import sys
input = sys.stdin.readline

n = int(input())

data = []
for _ in range(n):
    data.append(int(input()))

answer = 0
for i in range(n-1, 0, -1):
    if data[i] <= data[i-1]:
        gap = abs(data[i] - data[i-1]) + 1
        answer += gap
        data[i-1] -= gap
print(answer)

File: 백준/Silver/2847. 게임을 만든 동준이/게임을 만든 동준이.js
const fs = require('fs');
const [n, ...inputs] = fs.readFileSync('/dev/stdin').toString().trim().split('\n');
[...data] = inputs.map(input => parseInt(input));

let answer = 0;
for (let i = n - 2; i > -1; i--) {
  if (data[i] >= data[i + 1]) {
    const gap = Math.abs(data[i] - data[i + 1]) + 1;
    answer += gap;
    data[i] -= gap;
  }
}
console.log(answer);